### ISSUE TITLE
Short-term focus loss in a Quake mode

### ISSUE THREAD

**Issue by katoquro** [gnunn1/tilix#1647]
Hello!
I'm using [autokey](https://github.com/autokey/autokey) with tilix and when I hit an autokey shortcut it catches focus for a moment. With the option "Hide window when focus is lost" it is a bit inconvenient. 

If there were an option to set time interval within which returned focus doesn't considered as lost I could use tilix with autokey and autohiding feature.
Thanks!

**Comment by gnunn1** [comment]
Isn't that what the existing "Delay hiding by (ms)" option already does in Quake options or am I mis-understanding?

**Comment by katoquro** [comment]
Thanks for the quick reply!
I've tried setting it as max as possible (1000) and after my autokey macro done (I think less than 1 sec) tilix closes after that time. And I thought that option is about animation delay for the inevitable closing.
I use ubuntu 18.04 and I've tried updated to [1.8.9](https://launchpad.net/~webupd8team/+archive/ubuntu/terminix?field.series_filter=bionic) and have the same behaviour 

May be I can collect some debug info or event log?!

**Comment by katoquro** [comment]
I'm not sure but can it be [here](https://github.com/gnunn1/tilix/blob/552923400bada2f0a1960e4b9a29b8588d46be31/source/gx/tilix/appwindow.d#L1752)?  
I mean additional check if window is not under the focus.

Btw is there a way to enable fine grained logging for debug?
Thanks!

**Comment by gnunn1** [comment]
You can only enable fine-grained logging by building tilix from source code, i.e.

```dub build --config=trace```

And yes, you can try adding a check at that line to make sure the window doesn't have focus.

**Comment by curiousercreative** [comment]
@katoquro did you have any success with this? I'm also using Autokey and suffering :/

**Comment by katoquro** [comment]
@curiousercreative Yep. I've changed terminal to guake and added correspond functionality there on python )
